typescript/infra/config/environments/mainnet2/chains.ts

```diff
@@ -39,12 +41,21 @@ export const mainnetConfigs: ChainMap<ChainMetadata> = {
 };
 
 export type MainnetChains = keyof typeof mainnetConfigs;
-export const chainNames = Object.keys(mainnetConfigs) as MainnetChains[];
+export const supportedChainNames = Object.keys(
+  mainnetConfigs,
+) as MainnetChains[];
 export const environment = 'mainnet2';
 
-// Chains that we want to run agents for.
-export const agentChainNames = [
-  ...chainNames,
-  chainMetadata.solana.name,
-  chainMetadata.nautilus.name,
+const validatorChainNames = [
+  ...supportedChainNames,
+  'solanadevnet',
+  'proteustestnet',
 ];
+
+const relayerChainNames = validatorChainNames;
+
+export const agentChainNames: AgentChainNames = {
+  [Role.Validator]: validatorChainNames,
+  [Role.Relayer]: relayerChainNames,
+  [Role.Scraper]: supportedChainNames,
+};
```

Comment:
The contribution is essentially to use a `Record` type here keyed by the agent type, instead of an array